configure: Use string for tm_zone assignment

This matches what the actual sources do. Clang 16 and GCC 14 no longer
support converting ints to pointers implicitly, so the configure probe
always fails with these compilers.

Submitted upstream: <https://github.com/libimobiledevice/libplist/pull/249>

diff --git a/configure.ac b/configure.ac
index a9de50374c64849b..6ab01ae3a37312a9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -120,7 +120,7 @@ AC_CACHE_CHECK(for tm_zone in struct tm, ac_cv_struct_tm_zone,
       #include <time.h>
     ], [
       struct tm tm;
-      tm.tm_zone = 1;
+      tm.tm_zone = (char*)"UTC";
     ])],
     [ac_cv_struct_tm_zone=yes],
     [ac_cv_struct_tm_zone=no]
